src/main/java/entities/Budget.java

```diff
@@ -205,4 +216,36 @@ public double getTotalCost() {
         }
         return expenses;
     }
+
+    @Override
+    public void vetoableChange(PropertyChangeEvent evt) throws PropertyVetoException {
+        switch (evt.getPropertyName()) { // TODO: Is the presence of this switch statement a code smell? Is it needed>
```

Comment:
Is the presence of this switch statement a code smell? Is it needed?

Reply:
It isn't a code smell because it is for choosing which subclass to create (like in the Simple Factory design pattern).